great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -121,18 +120,27 @@ def _query_for_year_and_month(
     return {"year": year, "month": month}
 
 
+SQL_DATETIME_COUNTER = 0
+
+
 def _get_sql_datetime_range(
-    conn: sqlalchemy.engine.base.Connection, table_name: str, col_name: str
+    conn: sqlalchemy.engine.base.Connection, data_from: str, col_name: str
 ) -> DatetimeRange:
-    q = f"select min({col_name}), max({col_name}) from {table_name}"
+    global SQL_DATETIME_COUNTER
+    from_name = f"_gx_sql_datetime_range_data_{SQL_DATETIME_COUNTER}"
+    SQL_DATETIME_COUNTER += 1
+    q = f"select min({col_name}), max({col_name}) from ({data_from}) as {from_name}"
     min_max_dt = list(conn.execute(q))[0]
+    if min_max_dt[0] is None or min_max_dt[1] is None:
+        raise SQLDatasourceError(
+            f"Data date range can not be determined for the query: {q}. The returned range was {min_max_dt}."
+        )
     return DatetimeRange(min=min_max_dt[0], max=min_max_dt[1])
 
 
-class TableAsset(DataAsset):
+class SQLAsset(DataAsset):
```

Comment:
1. Is the class meant to be instantiated?
2. Will a user ever have a config file with `type: sqlasset`?

If the answer is no, then it probably doesn't need to be registered by the `MetaDatasource`.
You could make this `_SQLAsset(DataAsset)`: and this registration step would be skipped.

https://github.com/great-expectations/great_expectations/blob/1e91b803778309a4d18c27fadb156148b7109760/great_expectations/experimental/datasources/sources.py#L127-L132

However, it might be better to be more explicit about this and have some `ClassVar` `_SKIP_ASSET_REGISTRATION` attribute that gets checked.